include/rxmesh/kernels/attribute.cuh

```diff
@@ -94,6 +94,86 @@ __launch_bounds__(blockSize) __global__
     }
 }
 
+struct CustomMaxPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value > a.value) ? b : a;
+    }
+};
+
+struct CustomMinPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value < a.value) ? b : a;
+    }
+};
+template <class T, uint32_t blockSize, typename HandleT>
+__launch_bounds__(blockSize) __global__
+    void arg_max_kernel(const Attribute<T, HandleT> X,
+                    bool                        is_min,
+                    const uint32_t              num_patches,
+                    const uint32_t              num_attributes,
+                    cub::KeyValuePair<int, T>*  d_block_output,
+                    uint32_t                    attribute_id)
+{
+    using LocalT = typename HandleT::LocalT;
+
+    assert(X.get_num_attributes() == 1); //we can only take arg max for a scalar attribute
+
+    uint32_t p_id = blockIdx.x;
+    if (p_id < num_patches) {
+        const uint16_t element_per_patch = X.size(p_id);
+        cub::KeyValuePair<int, T> thread_val;
```

Comment:
I think the `KeyValuePair` should the HandleT and the value, i.e., `cub::KeyValuePair<HandleT, T>`.  This way `current_pair` will be something like that 

```
HandleT handle(p_id, i);
cub::KeyValuePair<int, T> current_pair(handle, X(p_id, i, attribute_id));
```